### PR TITLE
ENT-3884: refactor ``handle_user_post_save`` signal to account for admin permissions

### PR DESCRIPTION
**Pull request by adamstankiewicz**
https://openedx.atlassian.net/browse/ENT-3884

Related PR: https://github.com/edx/frontend-app-admin-portal/pull/421

Previously, admin permissions were granted to the user based on the creation/update of an `EnterpriseCustomerUser` record through a Django signal. However, if a user is already linked to an `EnterpriseCustomer`, there is no creation/update of an `EnterpriseCustomerUser` record so the admin permissions were never granted on their next login.

Instead, the granting of admin permissions was moved to coincide whenever a user logs in instead so that we will always grant the correct access roles on each login.

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [x] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [x] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

**Review by kellmarie — APPROVED** [review]

File: enterprise/models.py [hunk 1]
```diff
@@ -1107,6 +1110,74 @@ class Meta:
...
         ordering = ['created']
 
+    @classmethod
+    def link_pending_enterprise_user(cls, pending_ecu, is_user_created, user):
+        """
+        Link a PendingEnterpriseCustomerUser to the appropriate EnterpriseCustomer by
+        creating a EnterpriseCustomerUser record.
+
+        Arguments:
+            pending_ecu: a PendingEnterpriseCustomerUser instance
```
Comment (brittneyexline) [inline enterprise/models.py:1120]: why pass in pending_ecu as an argument when this method is in its class? why not just call this method on the pending_ecu?

**Review by brittneyexline — APPROVED** [review]
i think everything here makes sense, but now i understand why this is somewhat risky. i would recommend outlining a few manual test cases that can be performed in stage/prod to smoke test with. not sure if there is any particular dashboard that would be worth monitoring to see that there is no impact? in any case, i trust your judgement if anything goes awry 😄

**Comment by adamstankiewicz** [comment]
> i think everything here makes sense, but now i understand why this is somewhat risky. i would recommend outlining a few manual test cases that can be performed in stage/prod to smoke test with. not sure if there is any particular dashboard that would be worth monitoring to see that there is no impact? in any case, i trust your judgement if anything goes awry 😄

Most manual test cases are nicely outlined in the ticket: https://openedx.atlassian.net/browse/ENT-3884

I will document a couple other edge cases to verify in the PR description shortly 👍 Not sure if there is a dashboard worth monitoring here; I will try to see if there is one and monitor it once this is on stage/prod.
